test/functional/lazy-relations/entity/Category.ts

```diff
@@ -24,4 +30,193 @@ export class Category {
     @OneToMany(type => Post, post => post.twoSideCategory)
     twoSidePosts2: Promise<Post[]>;
 
+    // ManyToMany with named properties
+    @ManyToMany(type => PostNamedTable, post => post.categoriesNamedTable)
+    postsNamedTable: Promise<PostNamedTable[]>;
+
+    @ManyToMany(type => PostNamedColumn, post => post.categoriesNamedColumn)
+    postsNamedColumn: Promise<PostNamedColumn[]>;
+
+    @ManyToMany(type => PostNamedAll, post => post.categoriesNamedAll)
+    postsNamedAll: Promise<PostNamedAll[]>;
+
+    // OneToMany with named properties
+    @OneToMany(type => PostNamedTable, post => post.categoryNamedTable)
+    onePostsNamedTable: Promise<PostNamedTable[]>;
+
+    @OneToMany(type => PostNamedColumn, post => post.categoryNamedColumn)
+    onePostsNamedColumn: Promise<PostNamedColumn[]>;
+
+    @OneToMany(type => PostNamedAll, post => post.categoryNamedAll)
+    onePostsNamedAll: Promise<PostNamedAll[]>;
+
+    // OneToOne with named properties
+    @OneToOne(type => PostNamedTable, post => post.oneCategoryNamedTable)
+    onePostNamedTable: Promise<PostNamedTable>;
+
+    @OneToOne(type => PostNamedColumn, post => post.oneCategoryNamedColumn)
+    onePostNamedColumn: Promise<PostNamedColumn>;
+
+    @OneToOne(type => PostNamedAll, post => post.oneCategoryNamedAll)
+    onePostNamedAll: Promise<PostNamedAll>;
+
+}
+
+@Entity("s_category", {
+    orderBy: {
+        id: "ASC",
+    }
+})
+export class CategoryNamedTable {
```

Comment:
Dont put two entities in same class. Bad practice.

Reply:
What do you mean 2 entities in same class? did you meant 2 entities in same file?

Reply:
yes sorry, I meant 2 entities in same file

Reply:
Check 3rd comment on why i tried to put them there.